onnxruntime/core/mlas/lib/qsoftmax_kernel_avx2.cpp

```diff
@@ -0,0 +1,267 @@
+/*++
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+
+Licensed under the MIT License.
+
+Module Name:
+
+    qsoftmax_kernel_avx2.cpp
+
+Abstract:
+
+    This module implements the float/quantized n-bit integer matrix
+    multiplication kernels for x64 avx2.
+
+--*/
+
+#include "mlasi.h"
+
+static uint8_t reduce_max_u8_avx2(const uint8_t* data, size_t size) {
+  __m256i max_vec = _mm256_set1_epi8(0);
+
+  size_t i;
+  for (i = 0; i + 32 <= size; i += 32) {
+    __m256i vec = _mm256_loadu_si256((const __m256i*)(data + i));
+    max_vec = _mm256_max_epu8(max_vec, vec);
+  }
+
+  // Now reduce the 256-bit max_vec into a single max value
+  // First, split the 256-bit vector into two 128-bit halves and compute the max
+  // between them
+  __m128i max_128 = _mm_max_epu8(_mm256_castsi256_si128(max_vec), _mm256_extracti128_si256(max_vec, 1));
+
+  // Further reduce the 128-bit vector to a scalar
+  // Extract the upper 64-bit part and compute the max
+  max_128 = _mm_max_epu8(max_128, _mm_srli_si128(max_128, 8));
+  // Extract the upper 32-bit part and compute the max
+  max_128 = _mm_max_epu8(max_128, _mm_srli_si128(max_128, 4));
+  // Extract the upper 16-bit part and compute the max
+  max_128 = _mm_max_epu8(max_128, _mm_srli_si128(max_128, 2));
+  // Extract the upper 8-bit part and compute the max
+  max_128 = _mm_max_epu8(max_128, _mm_srli_si128(max_128, 1));
+
+  // Extract the final max value
+  uint8_t max_value = static_cast<uint8_t>(_mm_extract_epi8(max_128, 0));
+
+  for (; i < size; ++i) {
+    if (data[i] > max_value) {
+      max_value = data[i];
+    }
+  }
+
+  return max_value;
+}
+
+int8_t reduce_max_i8_avx2(const int8_t* data, size_t size) {
```

Comment:
Why you here doesn't follow the reduce_max_u8_avx2 above to use _mm_max_epi8

Reply:
Improved